### ISSUE TITLE
Duplicate files handling

### ISSUE THREAD

**Issue by fhammerschmidt** [rolandpeelen/rewatch#14]
Thank you very much for trying to tackle the ol' ReScript workspace repo problem.

I was excited to try it out, but encountered an error as soon as a ran `yarn rewatch build .`. It gave me
```
[2023-03-22T10:38:51Z ERROR rewatch::build] Duplicate files found for module: ChatAttachmentView
[2023-03-22T10:38:51Z ERROR rewatch::build] file 1: /Users/florian-cca/workspace/alert/node_modules/app-web/src/chat/ChatAttachmentView.res
[2023-03-22T10:38:51Z ERROR rewatch::build] file 2: /Users/florian-cca/workspace/alert/node_modules/app-mobile/src/chat/ChatAttachmentView.res
```

It seems like you still require unique filenames over all packages? It would be cool if that was supported, then rewatch would basically be a drop-in replacement since otherwise our repo structures are pretty similar to your testrepo.

**Comment by rolandpeelen** [comment]
Could you try enabling [namespacing](https://rescript-lang.org/docs/manual/latest/build-configuration) to see if it works then?

**Comment by rolandpeelen** [comment]
@fhammerschmidt - Did you get this to work on your side? I think the bits you are referring too may be something orthogonal to the build system. Perhaps Rescripts bs-build is a bit more flexible in this regard?

**Comment by fhammerschmidt** [comment]
I did not. But I found out that the above error only happens with dependencies that use the namespace field themselves. They all also set the namespace manually as a string. I tried to get rid of those errors by changing the namespace field in the bsconfig of the respective dependencies to the module name that was found. So e.g. in rescript-react-native there is 

```json
namespace: "react-native"
´´´ 

which I changed to 
```json
namespace: "React-native"
´´´ 

and that got rid of the error.

I feel like rewatch somewhere unnecessarily uppercases some module name somewhere.
